### PR TITLE
Use retry with exponential backoff

### PR DESCRIPTION
**Pull request by jcmorrow**
To try and fix S3 ECONNRESET errors

Fixes https://linear.app/replay/issue/BAC-3304/http-uploads-for-recordings-failing

### CODE REVIEW

File: packages/replay/src/main.ts [hunk 1]
```diff
@@ -394,7 +394,7 @@ async function doUploadRecording(
     recordingId,
   });
 
-  await client.uploadRecording(recording.path!, uploadLink, size);
+  await exponentialBackoffRetry(() => client.uploadRecording(recording.path!, uploadLink, size));
```
Comment (ryanjduffy) [inline packages/replay/src/main.ts:397]: Can you add a `debug()` entry within `client.uploadRecording()` so we have visibility into retries?

**Review by ryanjduffy — CHANGES_REQUESTED** [review]
Good plan! One note on some observability but otherwise let's give this a go!

File: packages/replay/src/main.ts (hunk 1 above)
Comment (jcmorrow) [reply packages/replay/src/main.ts:397]: yeah good call, I'll also record how long it took to fail, because I feel like that might tell us something.

Comment (jcmorrow) [reply packages/replay/src/main.ts]: Looks like `debug` does the time offset automatically! ❤️

**Pull request merged**